pkgs/top-level/all-packages.nix

```diff
@@ -217,7 +217,11 @@ with pkgs;
   } (writeScript "ensure-newer-sources-hook.sh" ''
       postUnpackHooks+=(_ensureNewerSources)
       _ensureNewerSources() {
-        '${findutils}/bin/find' "$sourceRoot" \
+        local r=$sourceRoot
+        # Avoid passing option-looking directory to find. The example is diffoscope-269:
+        #   https://salsa.debian.org/reproducible-builds/diffoscope/-/issues/378
+        [[ $r == -* ]] && r="./$r"
+        '${findutils}/bin/find' "$r" \
```

Comment:
Isn’t `find` part of the stdenv?

Reply:
I did not change the `find` source location in this patch. But looking at https://pubs.opengroup.org/onlinepubs/9699919799/utilities/find.html `POSIX` does not document `-newermt` option. Thus something needs to make sure it's `GNU findutils`.  The hook seems to apply to handle only `.zip` files, thus maybe it's not that important WRT closure size and extra dependencies.

Yes, today `findutils` is currently pulled into `stdenv` via `pkgs/stdenv/generic/common-path.nix`. I'm not sure if `stdenv` is guaranteed to have `GNU find` implementation (or any `find` implementation) in future.